spec/index.bs

```diff
@@ -8075,16 +8078,39 @@ dictionary GPUFragmentState
                     : {{GPUTextureSampleType/"uint"}}
                     :: |output| must have an unsigned integer scalar type.
                 </dl>
-            - If |colorState|.{{GPUColorTargetState/blend}} is [=map/exist|provided=] and
-                |colorState|.{{GPUColorTargetState/blend}}.{{GPUBlendState/color}}.{{GPUBlendComponent/srcFactor}}
-                or .{{GPUBlendComponent/dstFactor}} uses the source alpha
-                (is any of {{GPUBlendFactor/"src-alpha"}}, {{GPUBlendFactor/"one-minus-src-alpha"}},
-                or {{GPUBlendFactor/"src-alpha-saturated"}}), then:
-                - |output| must have an alpha channel (that is, it must be a vec4).
+            - If |colorState|.{{GPUColorTargetState/blend}} is [=map/exist|provided=]:
+                - If |colorState|.{{GPUColorTargetState/blend}}.{{GPUBlendState/color}}.{{GPUBlendComponent/srcFactor}} or
+                    |colorState|.{{GPUColorTargetState/blend}}.{{GPUBlendState/color}}.{{GPUBlendComponent/dstFactor}} or
+                    |colorState|.{{GPUColorTargetState/blend}}.{{GPUBlendState/alpha}}.{{GPUBlendComponent/srcFactor}} or
+                    |colorState|.{{GPUColorTargetState/blend}}.{{GPUBlendState/alpha}}.{{GPUBlendComponent/dstFactor}}
+                    uses the second input of the corresponding blending unit (is any of
+                    {{GPUBlendFactor/"src1"}}, {{GPUBlendFactor/"one-minus-src1"}},
+                    {{GPUBlendFactor/"src1-alpha"}}, {{GPUBlendFactor/"one-minus-src1-alpha"}}), then:
+                        - Set |UseDualSourceBlending| to `true`.
+                - If |colorState|.{{GPUColorTargetState/blend}}.{{GPUBlendState/color}}.{{GPUBlendComponent/srcFactor}}
+                    or |colorState|.{{GPUColorTargetState/blend}}.{{GPUBlendState/color}}.{{GPUBlendComponent/dstFactor}}
+                    uses the source alpha (is any of {{GPUBlendFactor/"src-alpha"}},
+                    {{GPUBlendFactor/"one-minus-src-alpha"}} or
+                    {{GPUBlendFactor/"src-alpha-saturated"}}) and the [=blend_src=] attribute of
+                    |output| is `0`, then:
+                        - |output| must have an alpha channel (that is, it must be a vec4).
 
             Otherwise, since there is no shader output for the attachment:
```

Comment:
Making line 8064 a loop makes this invalid, so I refactored this rule.

As an aside, this whole thing is getting so complicated that it may be preferable to rewrite the whole thing in terms of "Here is how you do the calculation; if any undefined values come out the end, then it's invalid" (not in this PR though).
But there were some specific rules that needed to be more strict than that, and this is one of them, which makes it harder.